client/src/partials/vouchers/complex.js

```diff
@@ -255,8 +264,84 @@ function ComplexJournalVoucherController(Vouchers, $translate, Accounts, Currenc
     // init sum debit and credit
     vm.sumDebit  = 0;
     vm.sumCredit = 0;
+
+    // init financial accounts
+    vm.financialAccount = [];
+
+    Cashbox.read(null, { detailed: 1 })
+    .then(function (list) {
+
+      list.forEach(function (item) {
+
+        if (!existInArray(item.account_id, vm.financialAccount)) {
+          vm.financialAccount.push(item.account_id);
+        }
+
+        if (!existInArray(item.transfer_account_id, vm.financialAccount)) {
+          vm.financialAccount.push(item.transfer_account_id);
+        }
+
+      });
+
+    })
+    .catch(Notify.handleError);
   }
 
+  /** check if an element exists in an array */
+  function existInArray(element, array) {
```

Comment:
This would be much better written as `array.indexOf(element) !== -1;`.  It's more standard javascript.